src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1293,6 +1299,161 @@ private function registerPropertyAccessConfiguration(array $config, ContainerBui
         ;
     }
 
+    private function registerAmqpConfiguration(array $config, ContainerBuilder $container)
+    {
+        $defaultConnectionName = null;
+        if (isset($config['default_connection'])) {
+            $defaultConnectionName = $config['default_connection'];
+        }
+
+        $match = false;
+        foreach ($config['connections'] as $name => $connection) {
+            $container
+                ->register("amqp.broker.$name", Broker::class)
+                ->addArgument($connection['url'])
+                ->addArgument($connection['queues'])
+                ->addArgument($connection['exchanges'])
+                ->setPublic(false)
+            ;
+            if (!$defaultConnectionName) {
+                $defaultConnectionName = $name;
+            }
+            if ($defaultConnectionName === $name) {
+                $match = true;
+            }
+        }
+
+        if (!$match) {
+            throw new \InvalidArgumentException(sprintf('The default_connection "%s" does not exist.', $defaultConnectionName));
+        }
+
+        $container->setAlias('amqp.broker', sprintf('amqp.broker.%s', $defaultConnectionName));
+    }
+
+    private function registerWorkerConfiguration(array $config, ContainerBuilder $container)
+    {
+        $fetchers = array();
+
+        foreach ($config['fetchers']['amqps'] as $name => $fetcher) {
+            if (isset($fetchers[$name])) {
+                throw new \InvalidArgumentException(sprintf('A fetcher named "%s" already exist.', $name));
+            }
+            if (isset($fetcher['connection'])) {
+                $connection = new Reference('amqp.broker.'.$fetcher['connection']);
+            } else {
+                $connection = new Reference('amqp.broker');
```

Comment:
The reference is invalid if `amqp` is not enabled, right? Does (or should) enabling `worker` require to enable `amqp`?